apps/backend/src/controllers/neighborhoods.ts

```diff
@@ -105,4 +107,102 @@ neighborhoodsRouter.post('/:id/join', middleware.userIdExtractorAndLoginValidato
   return res.status(201).send({ success: 'You have joined the neighborhood' });
 }));
 
+// Get a neighborhood's requests
+neighborhoodsRouter.get('/:id/requests', middleware.userIdExtractorAndLoginValidator, catchError(async (req: RequestWithAuthentication, res: Response) => {
+  const neighborhoodID = Number(req.params.id);
+  // LoginValidator ensures that loggedUserId is present
+  const loggedUserID = req.loggedUserId as number;
+
+  const isUserMemberOfNeighborhood: boolean = await neighborhoodServices
+    .isUserMemberOfNeighborhood(loggedUserID, neighborhoodID);
+
+  if (!isUserMemberOfNeighborhood) {
+    return res.status(400).send({ error: 'user is not a member of the neighborhood' });
+  }
+
+  const requests: RequestData[] = await neighborhoodServices
+    .getRequestsAssociatedWithNeighborhood(neighborhoodID);
+
+  return res.status(200).send(requests);
+}));
```

Comment:
Now I realize that theoretically we don't need this route at all. GET /neighborhoods/:id already extracts all the requests of a particular neighborhood if a user is logged in.
From a best practice point of view I think it's ok to have it (in case a third party wants to use our API), but I don't think we will be using it.